src/NuGet.Core/NuGet.ProjectManagement/Projects/FolderNuGetProject.cs

```diff
@@ -109,12 +103,12 @@ public override Task<IEnumerable<PackageReference>> GetInstalledPackagesAsync(Ca
                     await PackageExtractor.ExtractPackageAsync(
                         downloadResourceResult.PackageStream,
                         PackagePathResolver,
-                        nuGetProjectContext.PackageExtractionContext,
-                        PackageSaveMode,
+                        nuGetProjectContext.PackageExtractionContext ?? new PackageExtractionContext(),
```

Comment:
why wouldn't the project context have a default packageextractioncontext? Then all the ?? is not necessary. You can even go further and make it a get only property (or throw if it is set to null)